src/uct/ib/mlx5/rc/rc_mlx5_common.c

```diff
@@ -621,19 +621,18 @@ uct_rc_mlx5_dp_ordering_ooo_init(uct_rc_mlx5_iface_common_t *iface,
      * 
      * Fail at the following cases:
      * - Want to force configuration but can't force it
-     * - Want to force enable or force disable of adaptive routing but its unavailable
+     * - Want to force enable or force disable adaptive routing but it is unavailable
      * - Want to force disable adaptive routing but force enabling DDP
      */
 
     iface->config.dp_ordering = UCT_IB_MLX5_DP_ORDERING_IBTA;
+    min_forced_ordering       = UCT_IB_MLX5_DP_ORDERING_IBTA;
 
-    min_forced_ordering = (config->ddp_enable == UCS_YES) ?
-                                  UCT_IB_MLX5_DP_ORDERING_OOO_ALL :
-                                  UCT_IB_MLX5_DP_ORDERING_IBTA;
-    min_forced_ordering = (config->super.ar_enable == UCS_YES) ?
-                                  ucs_max(min_forced_ordering,
-                                          UCT_IB_MLX5_DP_ORDERING_OOO_RW) :
-                                  min_forced_ordering;
+    if (config->ddp_enable == UCS_YES) {
+        min_forced_ordering = UCT_IB_MLX5_DP_ORDERING_OOO_ALL;
+    } else if (config->super.ar_enable == UCS_YES) {
+        min_forced_ordering = UCT_IB_MLX5_DP_ORDERING_OOO_RW;
+    }
```

Comment:
add the one related sanity check here to reduce list of other conditions below?
`if (min_forced_ordering > dp_ordering_cap) {`